src/context/auth.jsx

```diff
@@ -1,168 +1,210 @@
-                username,
-                githubUsername,
-                mobile,
-                bio
-            )
-
-            if (res.status === 'success') {
-                navigate('/')
-                return
-            }
-            setError(ERR.PROFILE_CREATION_FAILED)
-        } catch (error) {
-            setError(ERR.PROFILE_CREATION_FAILED)
-        }
-    }
-
-    const value = {
-        token,
-        user,
-        onLogin: handleLogin,
-        onLogout: handleLogout,
-        onRegister: handleRegister,
-        onCreateProfile: handleCreateProfile,
-        error,
-        setError,
-        useClickOutside,
+      );
+
+      if (res.status === "success") {
+        navigate("/");
+        return;
+      }
+      setError(ERR.PROFILE_CREATION_FAILED);
+    } catch (error) {
+      setError(ERR.PROFILE_CREATION_FAILED);
     }
-
-    return <AuthContext.Provider value={value}>{children}</AuthContext.Provider>
-}
+  };
+
+  const value = {
+    token,
+    user,
+    onLogin: handleLogin,
+    onLogout: handleLogout,
+    onRegister: handleRegister,
+    onCreateProfile: handleCreateProfile,
+    error,
+    setError,
+    useClickOutside,
+  };
+
+  return <AuthContext.Provider value={value}>{children}</AuthContext.Provider>;
+};
 
 const ProtectedRoute = ({ children, disabledNav = false }) => {
-    const { token } = useAuth()
-    const location = useLocation()
+  const { token } = useAuth();
+  const location = useLocation();
 
-    if (!token) {
-        return <Navigate to="/login" replace state={{ from: location }} />
-    }
-
-    if (disabledNav) {
-        return (
-            <div className="container">
-                <Header />
-                <Navigation disabledNav />
-                <Modal />
-                {children}
-            </div>
-        )
-    }
+  if (!token) {
+    return <Navigate to="/login" replace state={{ from: location }} />;
+  }
 
+  if (disabledNav) {
     return (
-        <div className="container">
-            <Header />
-            <Navigation />
-            <Modal />
-            {children}
-        </div>
-    )
+      <div className="container">
+        <Header />
+        <Navigation disabledNav />
+        <Modal />
+        {children}
+      </div>
+    );
+  }
+
+  return (
+    <div className="container">
+      <Header />
+      <Navigation />
+      <Modal />
+      {children}
+    </div>
+  );
+};
+
+function validateEmail(email) {
+  const emailPattern = /^[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$/;
+
+  return !email || !emailPattern.test(email) ? false : true;
+}
+
+function validatePassword(password) {
+  const minLength = 8;
+  const hasUppercase = /[A-Z]/.test(password);
+  const hasNumber = /\d/.test(password);
+  const hasSpecialCharacter = /[!@#$%^&*(),.?":{}|<>]/.test(password);
+
+  if (
+    password.length < minLength ||
+    !hasUppercase ||
+    !hasNumber ||
+    !hasSpecialCharacter
+  ) {
+    return false;
+  }
+  return true;
```

Comment:
If you flip the conditional checks, you can just return the result of the evaluation:

```js
return password.length >= minLength && hasUppercase && hasNumber && hasSpecialCharacter
```